libopae/plugins/xfpga/event.c

```diff
@@ -468,32 +477,40 @@ STATIC fpga_result daemon_unregister_event(fpga_handle handle,
 					   fpga_event_type event_type)
 {
 	fpga_result result = FPGA_OK;
-
 	struct _fpga_handle *_handle = (struct _fpga_handle *)handle;
-	struct _fpga_token *_token = (struct _fpga_token *)_handle->token;
-
 	struct event_request req;
 	ssize_t n;
-	errno_t e;
+	fpga_properties prop = NULL;
+	uint64_t object_id = (uint64_t) -1;
 
 	if (_handle->fdfpgad < 0) {
 		FPGA_MSG("No fpgad connection");
 		return FPGA_INVALID_PARAM;
 	}
 
-	memset_s(&req, sizeof(req), 0);
-	req.type = UNREGISTER_EVENT;
-	req.event = event_type;
+	/* get the requestor's object ID */
+	result = xfpga_fpgaGetPropertiesFromHandle(handle, &prop);
+	if (result != FPGA_OK) {
+		FPGA_ERR("failed to get properties");
+		goto out_close_conn;
+	}
 
-	e = strncpy_s(req.device, sizeof(req.device), _token->sysfspath,
-		      sizeof(_token->sysfspath));
-	if (EOK != e) {
-		FPGA_ERR("strncpy_s failed");
-		result = FPGA_EXCEPTION;
+	result = fpgaPropertiesGetObjectID(prop, &object_id);
+	if (result != FPGA_OK) {
+		fpgaDestroyProperties(&prop);
+		FPGA_ERR("failed to get object ID");
 		goto out_close_conn;
 	}
 
-	req.device[sizeof(req.device) - 1] = '\0';
+	result = fpgaDestroyProperties(&prop);
+	if (result != FPGA_OK) {
+		FPGA_ERR("failed to destroy properties");
+		goto out_close_conn;
+	}
+
+	req.type = UNREGISTER_EVENT;
+	req.event = event_type;
+	req.object_id = object_id;
```

Comment:
I wonder if some of this can be refactored to one function to avoid some duplication...

Reply:
Did you mean to say similar to daemon_register_event?  Yeah, they are similar.  register checks for and opens the unix domain socket, whereas unregister assumes that it is already open.  I like having the register/unregister pairs, but if you feel strongly about it..

Reply:
really just the routine to get the object id... I don't feel too strongly about it